tests/test_subprocess.py

```diff
@@ -149,30 +171,48 @@ def test_context(fp, monkeypatch):
 
 
 @pytest.mark.parametrize("fake", [False, True])
-def test_basic_process(fp, fake):
+@path_or_str
+def test_basic_process(fp, fake, rtype, ptype):
     fp.allow_unregistered(not fake)
     if fake:
         fp.register(
-            [PYTHON, "example_script.py"],
+            [rtype(PYTHON), "example_script.py"],
             stdout=["Stdout line 1", "Stdout line 2"],
             stderr=None,
         )
 
-    process = subprocess.Popen(
-        [PYTHON, "example_script.py"],
-        cwd=os.path.dirname(__file__),
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
-    out, err = process.communicate()
+    if (
+        not fake
```

Comment:
Okay, I've changed the tests to verify exactly what happens, rather than skipping. I think what you are asking is you'd like the `not fake` condition above to be removed? That's a bit of a follow up (a feature that touches some of the same code, but is an independent improvement). This improvement specifically makes `Path("x")` match `"x"`, while before they only matched if you called the subprocess with the exact same object as the registration set. Having `Path("x")` throw the same error as the real subprocess on Win < 3.8 seems like another good addition.

Reply:
Yes, that's what I meant. There is no rush with this (unless you do need it fast?), so we can wait for you to come back from vacation or we can merge it as is now and work on this `TypeError` later (I could also work on it if you prefer). Let me know what you prefer, I'm OK with both.  
And have a nice vacation :)

Reply:
Minor preference to merge this, then work on TypeError, since it's a distinct feature (that happens to modify this test).

I'll see if I can work on it in the next couple of hours, otherwise it will likely wait or be something you work on. :)

Unless you are in a rush to release something, I'd like to test this out with scikit-build-core pulling the plugin from git once everything is in to see how the recents fixes works in a project.

Reply:
I'm not in a rush at all, I can wait for a notification from your side that all is OK and I should release a new version.